pkg/flow/logging/logger.go

```diff
@@ -23,6 +23,10 @@ type EnabledAware interface {
 type Logger struct {
 	inner io.Writer // Writer passed to New.
 
+	buffer       [][]interface{} // Store logs before correctly determine the log format
+	hasLogFormat bool            // Confirmation whether log format has been determined
+	bufferMut    sync.Mutex
```

Comment:
nit: as a good practice we try to put the mutex field above the fields that it protects

Reply:
I've addressed. PTAL